commit 058a746aed714504781c3955b8c5147433bf4020
Author: Emmanuel Hocdet <manu@gandi.net>
Date:   Thu Jan 16 14:41:36 2020 +0100

    BUG/MINOR: ssl: ssl_sock_load_ocsp_response_from_file memory leak
    
    "set ssl cert <filename.ocsp> <payload>" CLI command must free
    previous context.
    
    This patch should be backport to 2.1
    
    (cherry picked from commit 0667faebcf55562d86c30af63f36fe86ba58fff9)
    Signed-off-by: Christopher Faulet <cfaulet@haproxy.com>

diff --git a/src/ssl_sock.c b/src/ssl_sock.c
index 1fac2d905..2c19fa5b3 100644
--- a/src/ssl_sock.c
+++ b/src/ssl_sock.c
@@ -945,7 +945,12 @@ static int ssl_sock_load_ocsp_response_from_file(const char *ocsp_path, char *bu
 		ocsp_response = NULL;
 		goto end;
 	}
-
+	/* no error, fill ckch with new context, old context must be free */
+	if (ckch->ocsp_response) {
+		free(ckch->ocsp_response->area);
+		ckch->ocsp_response->area = NULL;
+		free(ckch->ocsp_response);
+	}
 	ckch->ocsp_response = ocsp_response;
 	ret = 0;
 end:
